api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -353,11 +334,25 @@ private User initializeUserIfNeeded() {
       request.getDocumentType(), request.getDocumentNumber()
     );
 
-    User user = userProvider.get();
-    user.setBlockscoreId(person.getId());
-    user.setBlockscoreVerificationIsValid(person.isValid());
-    userDao.save(user);
+    User user = userService.setIdVerification(person.getId(), person.isValid());
```

Comment:
Do we want/expect to re-run this even after a successful verification has already been performed?

Reply:
I'm not sure what the right UX is... it's not clear we want users to be able to attempt to re-verify themselves once they have done so successfully.

Anyway, at the moment if you call this method multiple times the last one wins.